File: 01. DataStructures/02. Stack/LeetCode Questions/18. Maximum Nesting Depth of the Parentheses #1614/solution.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack<char> a;
        int count=0;
        int max=0;
        for(int i=0;i<s.length();i++){
            if(s[i]=='('){
                a.push(s[i]);
                count++;
            }
            if(count>max){
                max=count;
            }
            if(s[i]==')'){
                a.pop();
                count--;
            }
        }
        return max;
    }
};

File: 01. DataStructures/02. Stack/LeetCode Questions/18. Maximum Nesting Depth of the Parentheses #1614/solution.js
var maxDepth = function(s) {
    let currunt = 0;
    let max = 0;
    
    for(let i = 0; i < s.length; i++) {
        if(s[i] == "(") {
            currunt += 1;
            max = Math.max(max, currunt);
        } else if (s[i] == ")"){
            currunt -= 1;
        }
    }
    
    return max;
};
maxDepth("8*((1*(5+6))*(8/6))")
